### ISSUE TITLE
JSON RPC calls flooding logs

### ISSUE THREAD

**Issue by WyseNynja** [NethermindEth/nethermind#1827]
I am wanting to experiment with nodes other than geth. I setup a nethermind node on my server and it synced quickly and seemed to be working very well. However, when I try to run `brownie test` against it, it gets back invalid data for `eth_getcode` and then I get a HUGE amount of errors that do not stop even though `ganache-cli` and `brownie` have both stopped.

```
Attaching to ethereum_nethermind_1
nethermind_1       | 2020-05-03 20:06:51.7939|Peer pool size is: 4816 
nethermind_1       | 2020-05-03 20:06:51.8321|Peer pool size is: 4917 
nethermind_1       | 2020-05-03 20:06:51.8432|Sync peers - Initialized: 5 | All: 7 | Max: 100
nethermind_1       |    [Peer|eth65|10313344|MultiGeth/v1.9.10-stable-1f8eefa1/linux-amd64/go1.13.8|144.217.66.227:30745][][][0]
nethermind_1       |    [Peer|eth63|9995249|Parity-Ethereum/v2.6.7-beta-b463487-20191231/x86_64-linux-gnu/rustc1.40.0|23.253.234.56:30303][][][0]
nethermind_1       |    [Peer|eth65|9995249|Geth/v1.9.13-stable-cbc4ac26/windows-amd64/go1.14.2|10.11.12.16:30303][][][0]
nethermind_1       |    [Peer|eth65|9775576|Geth/v1.9.13-stable-cbc4ac26/darwin-amd64/go1.13.3|24.4.6.65:30303][][][0]
nethermind_1       |    [Peer|eth63|9199999|Parity-Ethereum/v2.6.6-beta-5162bc2c8-20191205/x86_64-linux-gnu/rustc1.39.0|174.21.172.63:30303][][][0] 
nethermind_1       | 2020-05-03 20:06:51.8602|Peer pool size is: 5018 
nethermind_1       | 2020-05-03 20:06:51.8863|Peer pool size is: 5119 
nethermind_1       | 2020-05-03 20:06:51.9122|Peer pool size is: 5220 
nethermind_1       | 2020-05-03 20:06:51.9365|Peer pool size is: 5321 
nethermind_1       | 2020-05-03 20:06:51.9672|Peer pool size is: 5422 
nethermind_1       | 2020-05-03 20:06:52.0184|Peer pool size is: 5523 
nethermind_1       | 2020-05-03 20:06:52.0595|Peer pool size is: 5624 
nethermind_1       | 2020-05-03 20:06:52.1014|Peer pool size is: 5725 
nethermind_1       | 2020-05-03 20:06:52.1488|Peer pool size is: 5826 
nethermind_1       | 2020-05-03 20:06:52.1864|Peer pool size is: 5927 
nethermind_1       | 2020-05-03 20:06:52.2458|Peer pool size is: 6028 
nethermind_1       | 2020-05-03 20:06:52.2750|Connected to 0 bootnodes, 1143 trusted/persisted nodes 
nethermind_1       | 2020-05-03 20:06:58.9000|Building ethash cache for epoch 333 
nethermind_1       | 2020-05-03 20:07:01.0147|Peer pool size is: 6129 
nethermind_1       | 2020-05-03 20:07:03.1479|Cache for epoch 333 with size 60423872 nd seed 5d5674707d426883d592850443381021eb4738ee11e242760fcdad8295700bf8 built in 4247ms 
nethermind_1       | 2020-05-03 20:07:03.1626|Discovered a new block   9995250 20:06:26 (0x435b4d...7c7072), tx count: 141 mined by Spark Pool, sent by [Peer|eth63|9995250|Parity-Ethereum/v2.6.7-beta-b463487-20191231/x86_64-linux-gnu/rustc1.40.0|23.253.234.56:30303] 
nethermind_1       | 2020-05-03 20:07:04.9979|Initializing WebSockets for client: ''. 
nethermind_1       | 2020-05-03 20:07:05.0333|Executing JSON RPC call net_version with params  
nethermind_1       | 2020-05-03 20:07:05.0668|Executing JSON RPC call eth_getblockbynumber with params latest,False 
nethermind_1       | 2020-05-03 20:07:05.2966|Executing JSON RPC call eth_getcode with params 0x0000000000000000000000000000000000000000,0x0 
nethermind_1       | 2020-05-03 20:07:05.3038|Executing JSON RPC call eth_getbalance with params 0x0000000000000000000000000000000000000000,0x0 
nethermind_1       | 2020-05-03 20:07:05.3038|Executing JSON RPC call eth_gettransactioncount with params 0x0000000000000000000000000000000000000000,0x0 
nethermind_1       | 2020-05-03 20:07:06.0173|Executing JSON RPC call eth_getcode with params 0x6b175474e89094c44da98b954eedeac495271d0f,0x0 
nethermind_1       | 2020-05-03 20:07:06.4389|Executing JSON RPC call eth_getcode with params 0x0000000000b3f879cb30fe243b4dfee438691c04,0x0 
nethermind_1       | 2020-05-03 20:07:06.8255|Executing JSON RPC call eth_getcode with params 0xc0a47dfe034b400b47bdad5fecda2621de6c4d95,0x0 
nethermind_1       | 2020-05-03 20:07:07.2851|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4586|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4586|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4586|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4586|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4586|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4611|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4649|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4690|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4731|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:07.4731|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 

...

nethermind_1       | 2020-05-03 20:07:11.3570|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:11.3570|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
nethermind_1       | 2020-05-03 20:07:11.3610|Executing JSON RPC call eth_getcode with params 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,0x0 
^CERROR: Aborting.
```

Restarting the server ends the errors. They do not re-appear until I try to run brownie commands again.

I am connecting with websockets. I think maybe nethermind isn't seeing the client disconnect and so is trying to send the message over and over. That's just a guess though.

Let me know if you need more information or more logs. I can try not using websockets for some of this, but not all of it.

**Comment by tkstanczak** [comment]
@WyseNynja it may be that I broke the websockets code recently and it is relooping the same request - me or @matilote will try to test it tomorrow or so with brownie or truffle on websockets

**Comment by tkstanczak** [comment]
@WyseNynja we have some delay here but should be able to test it soon.

**Comment by tkstanczak** [comment]
still pending, sorry, but it is bubbling up the priority list

**Comment by tkstanczak** [comment]
@WyseNynja fixed and will prepare release soon, sorry it took so long

**Comment by tkstanczak** [comment]
Fixed in 1.8.33

**Comment by WyseNynja** [comment]
Thank you!
